common/clientFactory.go

```diff
@@ -53,25 +54,31 @@ func createClient[T, U any](callbacks newClientCallbacks[T, U], u string, credIn
 	switch credInfo.CredentialType {
 	case ECredentialType.OAuthToken():
 		if credInfo.OAuthTokenInfo.IsEmpty() {
-			credOpOptions.panicError(errors.New("invalid state, cannot get valid OAuth token information"))
+			err = errors.New("invalid state, cannot get valid OAuth token information")
+			break
 		}
-		tc, err := credInfo.OAuthTokenInfo.GetTokenCredential()
+		var tc azcore.TokenCredential
+		tc, err = credInfo.OAuthTokenInfo.GetTokenCredential()
 		if err != nil {
-			credOpOptions.panicError(fmt.Errorf("unable to get token credential due to reason (%s)", err.Error()))
+			err = fmt.Errorf("unable to get token credential due to reason (%s)", err.Error())
+			break
 		}
-		return callbacks.TokenCredential(u, tc, options)
+		client, err = callbacks.TokenCredential(u, tc, options)
 	case ECredentialType.Anonymous():
-		return callbacks.NoCredential(u, options)
+		client, err = callbacks.NoCredential(u, options)
 	default:
-		credOpOptions.panicError(fmt.Errorf("invalid state, credential type %v is not supported", credInfo.CredentialType))
-		return nil, fmt.Errorf("invalid state, credential type %v is not supported", credInfo.CredentialType)
+		err = fmt.Errorf("invalid state, credential type %v is not supported", credInfo.CredentialType)
 	}
+	if err != nil {
+		credOpOptions.panicError(err)
```

Comment:
PanicError will either call `panic` or `o.panic` depending on which one is available.

Typically, we try to avoid `panic` unless it's a sanity check; Realistically, `o.panic` would safely log the error (both in the job log and stdout) and then proceed to cleanly exit AzCopy.

So, I suppose the correct questions here are:

1. Are all of these errors sanity checks?
2. If not, How would we safely exit from here?

Reply:
This is a sanity check. The SDK returns an error on client construction but it is always nil in the codebase.

Reply:
So basically this will catch either client construction errors (which should always be nil) or wrong credential errors which shouldnt happen anyway